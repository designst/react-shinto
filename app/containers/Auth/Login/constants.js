import createConstant from 'utils/createConstant';

export const AUTH_LOGIN_REQUEST = createConstant(__filename, 'AUTH_LOGIN_REQUEST');
export const AUTH_LOGIN_SUCCESS = createConstant(__filename, 'AUTH_LOGIN_SUCCESS');
export const AUTH_LOGIN_FAILURE = createConstant(__filename, 'AUTH_LOGIN_FAILURE');
