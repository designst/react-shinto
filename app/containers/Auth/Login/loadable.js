/* @flow */

import loadable from 'loadable-components';

import Loading from 'components/Loading';
import ErrorDisplay from 'components/ErrorDisplay';

export default loadable(() => import(/* webpackChunkName: 'auth/login' */ './index'), {
  LoadingComponent: Loading,
  ErrorComponent: ErrorDisplay,
});
