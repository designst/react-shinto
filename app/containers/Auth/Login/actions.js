import { LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE } from './constants';

export const loginRequest = ({ username, password }) => ({
  type: LOGIN_REQUEST,
  username,
  password,
});

export const loginSuccess = data => ({
  type: LOGIN_SUCCESS,
  data,
});

export const loginFailure = error => ({
  type: LOGIN_FAILURE,
  error,
});
