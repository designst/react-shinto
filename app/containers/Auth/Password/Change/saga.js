import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_PASSWORD_CHANGE_REQUEST } from 'containers/Auth/Password/Change/constants';

import {
  passwordChangeSuccess,
  passwordChangeFailure,
} from 'containers/Auth/Password/Change/actions';

const logger = createLogger(__filename);

const authPasswordChangeUrl = process.env.SHINTO_AUTH_PASSWORD_CHANGE_API_ENDPOINT;

export function* authPasswordChange(action) {
  const { oldPassword, newPassword, newPasswordConfirm } = action.payload;

  const apiService = yield getContext('apiService');

  logger('Password Change');

  try {
    const response = yield call(apiService.post, authPasswordChangeUrl, {
      oldPassword,
      newPassword,
      newPasswordConfirm,
    });
    yield put(passwordChangeSuccess(response.data));
  } catch (err) {
    yield put(passwordChangeFailure(err));
  }
}

export default function* authPasswordChangeData() {
  yield takeLatest(AUTH_PASSWORD_CHANGE_REQUEST, authPasswordChange);
}
