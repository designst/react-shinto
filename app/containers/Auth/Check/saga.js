import { call, put, select, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { makeSelectToken } from 'containers/Auth/Check/selectors';
import { CHECK_AUTH_REQUEST } from 'containers/Auth/Check/constants';
import { checkAuthSuccess, checkAuthFailure } from 'containers/Auth/Check/actions';

const logger = createLogger(__filename);

const checkAuthUrl = process.env.SHINTO_AUTH_CHECK_API_ENDPOINT;

export function* checkAuth() {
  const token = yield select(makeSelectToken());

  const apiService = yield getContext('apiService');

  logger('Check Auth: %s', token);

  try {
    const data = yield call(apiService.get, checkAuthUrl, { token });
    yield put(checkAuthSuccess(data));
  } catch (err) {
    yield put(checkAuthFailure(err));
  }
}

export default function* checkAuthData() {
  yield takeLatest(CHECK_AUTH_REQUEST, checkAuth);
}
