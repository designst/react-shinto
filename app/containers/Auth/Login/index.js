import { compose } from 'redux';
import { connect } from 'react-redux';

import Login from 'components/Auth/Login';

import { loginRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: loginRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(Login);
