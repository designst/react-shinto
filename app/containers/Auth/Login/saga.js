import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGIN_REQUEST } from 'containers/Auth/Login/constants';
import { loginSuccess, loginFailure } from 'containers/Auth/Login/actions';

const logger = createLogger(__filename);

const authLoginUrl = process.env.SHINTO_AUTH_LOGIN_API_ENDPOINT;

export function* authLogin(username, password) {
  const apiService = yield getContext('apiService');

  logger('Login: %s', username);

  try {
    const response = yield call(apiService.post, authLoginUrl, { username, password });
    yield put(loginSuccess(response.data));
  } catch (err) {
    yield put(loginFailure(err));
  }
}

export default function* authLoginData() {
  yield takeLatest(AUTH_LOGIN_REQUEST, authLogin);
}
