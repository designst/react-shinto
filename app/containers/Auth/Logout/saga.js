import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_LOGOUT_REQUEST } from 'containers/Auth/Logout/constants';
import { logoutSuccess, logoutFailure } from 'containers/Auth/Logout/actions';

const logger = createLogger(__filename);

const authLogoutUrl = process.env.SHINTO_AUTH_LOGOUT_API_ENDPOINT;

export function* authLogout() {
  const apiService = yield getContext('apiService');

  logger('Logout');

  try {
    const response = yield call(apiService.get, authLogoutUrl);
    yield put(logoutSuccess(response.data));
  } catch (err) {
    yield put(logoutFailure(err));
  }
}

export default function* authLogoutData() {
  yield takeLatest(AUTH_LOGOUT_REQUEST, authLogout);
}
