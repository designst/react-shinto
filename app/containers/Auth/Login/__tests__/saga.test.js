import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import { authLogin } from 'containers/Auth/Login/saga';

describe('authLoginData', () => {
  let authLoginGenerator;

  beforeEach(() => {
    authLoginGenerator = authLogin();

    const selectDescriptor = authLoginGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authLoginSuccess action if it requests the data successfully', () => {
    const username = 'username';
    const password = 'password';

    return expectSaga(authLogin, username, password)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/login'
              ? {
                  data: {
                    token: 'token',
                    isAuthenticated: true,
                    isAuthenticating: false,
                  },
                }
              : {},
        }),
      })
      .withReducer(authReducer)
      .hasFinalState({
        token: 'token',
        isAuthenticated: true,
        isAuthenticating: false,
      })
      .run();
  });

  it('should dispatch the authLoginFailure action if it requests the data unsuccessfully', () => {
    const username = 'username';
    const password = 'password';

    return expectSaga(authLogin, username, password)
      .withReducer(authReducer)
      .run();
  });
});
