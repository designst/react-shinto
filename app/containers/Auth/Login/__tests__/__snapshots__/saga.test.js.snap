// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authLoginData should dispatch the authLoginFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "GET_CONTEXT": "apiService",
}
`;

exports[`authLoginData should dispatch the authLoginSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "GET_CONTEXT": "apiService",
}
`;
